uaclient/timer/tests/test_update_messaging.py

```diff
@@ -365,8 +368,19 @@ def test_update_motd_messages(
         write_file_calls,
         FakeConfig,
     ):
-        m_is_attached.return_value = mock.MagicMock(is_attached=attached)
-        m_get_contract_expiry_status.side_effect = contract_expiry_statuses
+
+        if contract_expiry_statuses:
+            m_is_attached.side_effect = [
+                mock.MagicMock(
+                    is_attached=attached,
+                    contract_status=contract_status.value,
+                    contract_remaining_days=contract_remaining_days,
+                )
+                for contract_status, contract_remaining_days in contract_expiry_statuses  # noqa
+            ]
+        else:
+            m_is_attached.return_value = mock.MagicMock(is_attached=attached)
+
```

Comment:
I'd rather re-write the parametrization to provide a value for `m_is_attached.side_effect` always rather than add transformation logic in the test heere